{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Download complete.\n"
     ]
    }
   ],
   "source": [
    "# Download the list of Clinvar XML releases\n",
    "import requests\n",
    "\n",
    "clinvar_xml_url = \"https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/\"\n",
    "output_file = \"ClinVarFullRelease.xml.gz\"\n",
    "\n",
    "# Stream download\n",
    "response = requests.get(clinvar_xml_url, stream=True)\n",
    "if response.status_code == 200:\n",
    "    with open(output_file, \"wb\") as f:\n",
    "        for chunk in response.iter_content(chunk_size=8192):\n",
    "            f.write(chunk)\n",
    "    print(\"Download complete.\")\n",
    "else:\n",
    "    print(f\"Download failed! HTTP status code: {response.status_code}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarFullRelease_2023-10.xml.gz\n",
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarFullRelease_2024-02.xml.gz\n",
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarFullRelease_2024-04.xml.gz\n",
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_00-latest.xml.gz\n",
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_2024-02.xml.gz\n",
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_2024-03.xml.gz\n",
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_2024-04.xml.gz\n",
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_2024-05.xml.gz\n",
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_2024-06.xml.gz\n",
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_2024-07.xml.gz\n",
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_2024-08.xml.gz\n",
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_2024-09.xml.gz\n",
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_2024-10.xml.gz\n",
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_2024-11.xml.gz\n",
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_2024-12.xml.gz\n",
      "https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_2025-01.xml.gz\n"
     ]
    }
   ],
   "source": [
    "# Parse the HTML Page to Extract File Links\n",
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "# URL of the ClinVar XML directory\n",
    "url = \"https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/\"\n",
    "\n",
    "# Fetch the HTML page\n",
    "response = requests.get(url)\n",
    "html_content = response.text\n",
    "\n",
    "# Parse the HTML content\n",
    "soup = BeautifulSoup(html_content, \"html.parser\")\n",
    "\n",
    "# Find all links to .gz files\n",
    "file_links = []\n",
    "for link in soup.find_all(\"a\"):\n",
    "    href = link.get(\"href\")\n",
    "    if href and href.endswith(\".xml.gz\"):\n",
    "        file_links.append(url + href)\n",
    "\n",
    "# Display extracted file links\n",
    "for file in file_links:\n",
    "    print(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloaded ClinVarFullRelease_2025-01.xml.gz\n"
     ]
    }
   ],
   "source": [
    "# Download the Latest ClinVar XML File\n",
    "import os\n",
    "\n",
    "# Choose the latest file (modify as needed)\n",
    "latest_file_url = \"https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarVCVRelease_2025-01.xml.gz\"\n",
    "output_file = \"ClinVarFullRelease_2025-01.xml.gz\"\n",
    "\n",
    "# Download the file\n",
    "response = requests.get(latest_file_url, stream=True)\n",
    "if response.status_code == 200:\n",
    "    with open(output_file, \"wb\") as f:\n",
    "        for chunk in response.iter_content(chunk_size=8192):\n",
    "            f.write(chunk)\n",
    "    print(f\"Downloaded {output_file}\")\n",
    "else:\n",
    "    print(f\"Failed to download file. HTTP Status Code: {response.status_code}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File size: 4154767604 bytes\n",
      "The file is correctly recognized as a GZIP file.\n"
     ]
    }
   ],
   "source": [
    "# Check that the XML file is correct\n",
    "import os\n",
    "\n",
    "file_path = \"ClinVarFullRelease_2025-01.xml.gz\"\n",
    "\n",
    "# Check the file size\n",
    "print(f\"File size: {os.path.getsize(file_path)} bytes\")\n",
    "\n",
    "# Check if it's a gzip file\n",
    "with open(file_path, \"rb\") as f:\n",
    "    signature = f.read(2)  # Read the first two bytes\n",
    "\n",
    "if signature == b'\\x1f\\x8b':  # GZIP signature\n",
    "    print(\"The file is correctly recognized as a GZIP file.\")\n",
    "else:\n",
    "    print(\"The file is NOT a GZIP file. The download may be corrupted.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import xml.etree.ElementTree as ET\n",
    "# import gzip\n",
    "# import pandas as pd\n",
    "\n",
    "# # Define XML file path\n",
    "# xml_file = \"ClinVarFullRelease_2025-01.xml.gz\"\n",
    "\n",
    "# # Function to extract Gene Symbol, VariationID, Name, and PubMed ID\n",
    "# def extract_clinvar_data(xml_file, max_records=None):\n",
    "#     extracted_data = []\n",
    "\n",
    "#     with gzip.open(xml_file, \"rt\", encoding=\"utf-8\") as f:\n",
    "#         context = ET.iterparse(f, events=(\"start\", \"end\"))\n",
    "#         _, root = next(context)  # Get root element\n",
    "\n",
    "#         record_count = 0\n",
    "#         for event, elem in context:\n",
    "#             if event == \"end\" and elem.tag == \"VariationArchive\":\n",
    "#                 variation_id = elem.attrib.get(\"VariationID\", \"N/A\")\n",
    "#                 variation_name = elem.attrib.get(\"VariationName\", \"N/A\")\n",
    "\n",
    "#                 # Extract Gene Symbol\n",
    "#                 gene_elem = elem.find(\".//Gene\")\n",
    "#                 gene_symbol = gene_elem.attrib.get(\"Symbol\", \"N/A\") if gene_elem is not None else \"N/A\"\n",
    "\n",
    "#                 # Extract PubMed ID\n",
    "#                 citation_elem = elem.find(\".//Citation/ID[@Source='PubMed']\")\n",
    "#                 pubmed_id = citation_elem.text if citation_elem is not None else \"N/A\"\n",
    "\n",
    "#                 # Append data to list\n",
    "#                 extracted_data.append({\n",
    "#                     \"VariationID\": variation_id,\n",
    "#                     \"GeneSymbol\": gene_symbol,\n",
    "#                     \"Name\": variation_name,\n",
    "#                     \"PubMedID\": pubmed_id\n",
    "#                 })\n",
    "\n",
    "#                 record_count += 1\n",
    "#                 if max_records and record_count >= max_records:\n",
    "#                     break  # Stop after max_records for testing\n",
    "\n",
    "#                 # Free memory\n",
    "#                 elem.clear()\n",
    "\n",
    "#     return extracted_data\n",
    "\n",
    "# # Extract first 100 records for testing\n",
    "# clinvar_data = extract_clinvar_data(xml_file, max_records=10)\n",
    "\n",
    "# # Convert results to DataFrame\n",
    "# df = pd.DataFrame(clinvar_data)\n",
    "\n",
    "# # Save to CSV\n",
    "# csv_filename = \"ClinVar_Gene_Variants_2025-01.csv\"\n",
    "# df.to_csv(csv_filename, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<VariationArchive RecordType=\"classified\" VariationID=\"2\" VariationName=\"NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTAACTGTAAA (p.Arg27_Ile28delinsLeuLeuTer)\" VariationType=\"Indel\" Accession=\"VCV000000002\" Version=\"3\" NumberOfSubmissions=\"2\" NumberOfSubmitters=\"2\" DateLastUpdated=\"2022-04-25\" DateCreated=\"2017-01-30\" MostRecentSubmission=\"2021-05-16\">\n",
      "  <RecordStatus>current</RecordStatus>\n",
      "  <Species>Homo sapiens</Species>\n",
      "  <ClassifiedRecord>\n",
      "    <SimpleAllele AlleleID=\"15041\" VariationID=\"2\">\n",
      "      <GeneList>\n",
      "        <Gene Symbol=\"AP5Z1\" FullName=\"adaptor related protein complex 5 subunit zeta 1\" GeneID=\"9907\" HGNC_ID=\"HGNC:22197\" Source=\"submitted\" RelationshipType=\"within single gene\">\n",
      "          <Location>\n",
      "            <CytogeneticLocation>7p22.1</CytogeneticLocation>\n",
      "            <SequenceLocation Assembly=\"GRCh38\" AssemblyAccessionVersion=\"GCF_000001405.38\" AssemblyStatus=\"current\" Chr=\"7\" Accession=\"NC_000007.14\" start=\"4775623\" stop=\"4794397\" display_start=\"4775623\" display_stop=\"4794397\" Strand=\"+\" />\n",
      "            <SequenceLocation Assembly=\"GRCh37\" AssemblyAccessionVersion=\"GCF_000001405.25\" AssemblyStatus=\"previous\" Chr=\"7\" Accession=\"NC_000007.13\" start=\"4815261\" stop=\"4834025\" display_start=\"4815261\" display_stop=\"4834025\" Strand=\"+\" />\n",
      "          </Location>\n",
      "          <OMIM>613653</OMIM>\n",
      "        </Gene>\n",
      "      </GeneList>\n",
      "      <Name>NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTAACTGTAAA (p.Arg27_Ile28delinsLeuLeuTer)</Name>\n",
      "      <CanonicalSPDI>NC_000007.14:4781212:GGAT:TGCTGTAAACTGTAACTGTAAA</CanonicalSPDI>\n",
      "      <VariantType>Indel</VariantType>\n",
      "      <Location>\n",
      "        <CytogeneticLocation>7p22.1</CytogeneticLocation>\n",
      "        <SequenceLocation Assembly=\"GRCh38\" AssemblyAccessionVersion=\"GCF_000001405.38\" forDisplay=\"true\" AssemblyStatus=\"current\" Chr=\"7\" Accession=\"NC_000007.14\" start=\"4781213\" stop=\"4781216\" display_start=\"4781213\" display_stop=\"4781216\" variantLength=\"22\" positionVCF=\"4781213\" referenceAlleleVCF=\"GGAT\" alternateAlleleVCF=\"TGCTGTAAACTGTAACTGTAAA\" />\n",
      "        <SequenceLocation Assembly=\"GRCh37\" AssemblyAccessionVersion=\"GCF_000001405.25\" AssemblyStatus=\"previous\" Chr=\"7\" Accession=\"NC_000007.13\" start=\"4820844\" stop=\"4820847\" display_start=\"4820844\" display_stop=\"4820847\" variantLength=\"22\" positionVCF=\"4820844\" referenceAlleleVCF=\"GGAT\" alternateAlleleVCF=\"TGCTGTAAACTGTAACTGTAAA\" />\n",
      "      </Location>\n",
      "      <OtherNameList>\n",
      "        <Name>AP5Z1, 4-BP DEL/22-BP INS, NT80</Name>\n",
      "      </OtherNameList>\n",
      "      <HGVSlist>\n",
      "        <HGVS Assembly=\"GRCh37\" Type=\"genomic, top-level\">\n",
      "          <NucleotideExpression sequenceAccessionVersion=\"NC_000007.13\" sequenceAccession=\"NC_000007\" sequenceVersion=\"13\" change=\"g.4820844_4820847delinsTGCTGTAAACTGTAACTGTAAA\" Assembly=\"GRCh37\">\n",
      "            <Expression>NC_000007.13:g.4820844_4820847delinsTGCTGTAAACTGTAACTGTAAA</Expression>\n",
      "          </NucleotideExpression>\n",
      "        </HGVS>\n",
      "        <HGVS Type=\"coding\">\n",
      "          <NucleotideExpression sequenceAccessionVersion=\"NM_001364858.1\" sequenceAccession=\"NM_001364858\" sequenceVersion=\"1\" change=\"c.-202_-199delinsTGCTGTAAACTGTAACTGTAAA\">\n",
      "            <Expression>NM_001364858.1:c.-202_-199delinsTGCTGTAAACTGTAACTGTAAA</Expression>\n",
      "          </NucleotideExpression>\n",
      "          <MolecularConsequence ID=\"SO:0001623\" Type=\"5 prime UTR variant\" DB=\"SO\" />\n",
      "        </HGVS>\n",
      "        <HGVS Type=\"coding\">\n",
      "          <NucleotideExpression sequenceAccessionVersion=\"LRG_1247t1\" sequenceAccession=\"LRG_1247t1\" change=\"c.80_83delinsTGCTGTAAACTGTAACTGTAAA\">\n",
      "            <Expression>LRG_1247t1:c.80_83delinsTGCTGTAAACTGTAACTGTAAA</Expression>\n",
      "          </NucleotideExpression>\n",
      "          <ProteinExpression sequenceAccessionVersion=\"LRG_1247p1\" sequenceAccession=\"LRG_1247p1\" change=\"p.Arg27_Ile28delinsLeuLeuTer\">\n",
      "            <Expression>LRG_1247p1:p.Arg27_Ile28delinsLeuLeuTer</Expression>\n",
      "          </ProteinExpression>\n",
      "        </HGVS>\n",
      "        <HGVS Type=\"genomic\">\n",
      "          <NucleotideExpression sequenceAccessionVersion=\"LRG_1247\" sequenceAccession=\"LRG_1247\" change=\"g.10583_10586delinsTGCTGTAAACTGTAACTGTAAA\">\n",
      "            <Expression>LRG_1247:g.10583_10586delinsTGCTGTAAACTGTAACTGTAAA</Expression>\n",
      "          </NucleotideExpression>\n",
      "        </HGVS>\n",
      "        <HGVS Type=\"coding\">\n",
      "          <NucleotideExpression sequenceAccessionVersion=\"NM_014855.3\" sequenceAccession=\"NM_014855\" sequenceVersion=\"3\" change=\"c.80_83delinsTGCTGTAAACTGTAACTGTAAA\" MANESelect=\"true\">\n",
      "            <Expression>NM_014855.3:c.80_83delinsTGCTGTAAACTGTAACTGTAAA</Expression>\n",
      "          </NucleotideExpression>\n",
      "          <ProteinExpression sequenceAccessionVersion=\"NP_055670.1\" sequenceAccession=\"NP_055670\" sequenceVersion=\"1\" change=\"p.Arg27_Ile28delinsLeuLeuTer\">\n",
      "            <Expression>NP_055670.1:p.Arg27_Ile28delinsLeuLeuTer</Expression>\n",
      "          </ProteinExpression>\n",
      "          <MolecularConsequence ID=\"SO:0001587\" Type=\"nonsense\" DB=\"SO\" />\n",
      "        </HGVS>\n",
      "        <HGVS Assembly=\"GRCh38\" Type=\"genomic, top-level\">\n",
      "          <NucleotideExpression sequenceAccessionVersion=\"NC_000007.14\" sequenceAccession=\"NC_000007\" sequenceVersion=\"14\" change=\"g.4781213_4781216delinsTGCTGTAAACTGTAACTGTAAA\" Assembly=\"GRCh38\">\n",
      "            <Expression>NC_000007.14:g.4781213_4781216delinsTGCTGTAAACTGTAACTGTAAA</Expression>\n",
      "          </NucleotideExpression>\n",
      "        </HGVS>\n",
      "        <HGVS Type=\"genomic\">\n",
      "          <NucleotideExpression sequenceAccessionVersion=\"NG_028111.1\" sequenceAccession=\"NG_028111\" sequenceVersion=\"1\" change=\"g.10583_10586delinsTGCTGTAAACTGTAACTGTAAA\">\n",
      "            <Expression>NG_028111.1:g.10583_10586delinsTGCTGTAAACTGTAACTGTAAA</Expression>\n",
      "          </NucleotideExpression>\n",
      "        </HGVS>\n",
      "        <HGVS Type=\"non-coding\">\n",
      "          <NucleotideExpression sequenceAccessionVersion=\"NR_157345.1\" sequenceAccession=\"NR_157345\" sequenceVersion=\"1\" change=\"n.173_176delinsTGCTGTAAACTGTAACTGTAAA\">\n",
      "            <Expression>NR_157345.1:n.173_176delinsTGCTGTAAACTGTAACTGTAAA</Expression>\n",
      "          </NucleotideExpression>\n",
      "          <MolecularConsequence ID=\"SO:0001619\" Type=\"non-coding transcript variant\" DB=\"SO\" />\n",
      "        </HGVS>\n",
      "      </HGVSlist>\n",
      "      <XRefList>\n",
      "        <XRef ID=\"CA215070\" DB=\"ClinGen\" />\n",
      "        <XRef Type=\"Allelic variant\" ID=\"613653.0001\" DB=\"OMIM\" />\n",
      "        <XRef Type=\"rs\" ID=\"397704705\" DB=\"dbSNP\" />\n",
      "      </XRefList>\n",
      "    </SimpleAllele>\n",
      "    <RCVList>\n",
      "      <RCVAccession Title=\"NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTAACTGTAAA (p.Arg27_Ile28delinsLeuLeuTer) AND Hereditary spastic paraplegia 48\" Accession=\"RCV000000012\" Version=\"5\">\n",
      "        <ClassifiedConditionList TraitSetID=\"2\">\n",
      "          <ClassifiedCondition DB=\"MedGen\" ID=\"C3150901\">Hereditary spastic paraplegia 48</ClassifiedCondition>\n",
      "        </ClassifiedConditionList>\n",
      "        <RCVClassifications>\n",
      "          <GermlineClassification>\n",
      "            <ReviewStatus>criteria provided, single submitter</ReviewStatus>\n",
      "            <Description SubmissionCount=\"2\">Pathogenic</Description>\n",
      "          </GermlineClassification>\n",
      "        </RCVClassifications>\n",
      "      </RCVAccession>\n",
      "    </RCVList>\n",
      "    <Classifications>\n",
      "      <GermlineClassification NumberOfSubmissions=\"2\" NumberOfSubmitters=\"2\" DateCreated=\"2017-01-30\" MostRecentSubmission=\"2021-05-16\">\n",
      "        <ReviewStatus>criteria provided, single submitter</ReviewStatus>\n",
      "        <Description>Pathogenic</Description>\n",
      "        <Citation Type=\"general\">\n",
      "          <ID Source=\"PubMed\">20613862</ID>\n",
      "        </Citation>\n",
      "        <ConditionList>\n",
      "          <TraitSet ID=\"2\" Type=\"Disease\" ContributesToAggregateClassification=\"true\">\n",
      "            <Trait ID=\"9580\" Type=\"Disease\">\n",
      "              <Name>\n",
      "                <ElementValue Type=\"Preferred\">Hereditary spastic paraplegia 48</ElementValue>\n",
      "                <XRef ID=\"MONDO:0013342\" DB=\"MONDO\" />\n",
      "              </Name>\n",
      "              <Name>\n",
      "                <ElementValue Type=\"Alternate\">Spastic paraplegia 48</ElementValue>\n",
      "              </Name>\n",
      "              <Name>\n",
      "                <ElementValue Type=\"Alternate\">Spastic paraplegia 48, autosomal recessive</ElementValue>\n",
      "                <XRef ID=\"Spastic+paraplegia+48%2C+autosomal+recessive/9323\" DB=\"Genetic Alliance\" />\n",
      "              </Name>\n",
      "              <Symbol>\n",
      "                <ElementValue Type=\"Alternate\">SPG48</ElementValue>\n",
      "                <XRef Type=\"MIM\" ID=\"613647\" DB=\"OMIM\" />\n",
      "              </Symbol>\n",
      "              <XRef ID=\"306511\" DB=\"Orphanet\" />\n",
      "              <XRef ID=\"C3150901\" DB=\"MedGen\" />\n",
      "              <XRef ID=\"MONDO:0013342\" DB=\"MONDO\" />\n",
      "              <XRef Type=\"MIM\" ID=\"613647\" DB=\"OMIM\" />\n",
      "            </Trait>\n",
      "          </TraitSet>\n",
      "        </ConditionList>\n",
      "      </GermlineClassification>\n",
      "    </Classifications>\n",
      "    <ClinicalAssertionList>\n",
      "      <ClinicalAssertion ID=\"20155\" SubmissionDate=\"2017-01-26\" DateLastUpdated=\"2017-01-30\" DateCreated=\"2013-04-04\">\n",
      "        <ClinVarSubmissionID localKey=\"613653.0001_SPASTIC PARAPLEGIA 48, AUTOSOMAL RECESSIVE\" title=\"AP5Z1, 4-BP DEL/22-BP INS, NT80_SPASTIC PARAPLEGIA 48, AUTOSOMAL RECESSIVE\" />\n",
      "        <ClinVarAccession Accession=\"SCV000020155\" DateUpdated=\"2017-01-30\" DateCreated=\"2013-04-04\" Type=\"SCV\" Version=\"3\" SubmitterName=\"OMIM\" OrgID=\"3\" OrganizationCategory=\"resource\" />\n",
      "        <RecordStatus>current</RecordStatus>\n",
      "        <Classification DateLastEvaluated=\"2010-06-29\">\n",
      "          <ReviewStatus>no assertion criteria provided</ReviewStatus>\n",
      "          <GermlineClassification>Pathogenic</GermlineClassification>\n",
      "        </Classification>\n",
      "        <Assertion>variation to disease</Assertion>\n",
      "        <ObservedInList>\n",
      "          <ObservedIn>\n",
      "            <Sample>\n",
      "              <Origin>germline</Origin>\n",
      "              <Species>human</Species>\n",
      "              <AffectedStatus>not provided</AffectedStatus>\n",
      "            </Sample>\n",
      "            <Method>\n",
      "              <MethodType>literature only</MethodType>\n",
      "            </Method>\n",
      "            <ObservedData>\n",
      "              <Attribute Type=\"Description\">In 2 French sibs with autosomal recessive spastic paraplegia-48 (SPG48; 613647), Slabicki et al. (2010) identified a homozygous complex insertion/deletion mutation in exon 2 of the KIAA0415 gene. The mutation comprised a 4-bp deletion (80del4) and a 22-bp insertion (84ins22), resulting in a frameshift and premature stop codon following residue 29. The insertion was found to be an imperfect quadruplication of a sequence, suggesting DNA polymerase slippage during DNA synthesis as the pathogenetic mechanism. The patients presented with progressive spastic paraplegia associated with urinary incontinence from ages 50 and 49 years, respectively. One had a normal cerebral MRI, whereas the other had spinal hyperintensities in the cervical spine. The unaffected parents were not known to be consanguineous, but they originated from 2 neighboring villages. The mutation was not found in 156 Caucasian or 242 North African control chromosomes. Studies of lymphoblastoid cells derived from 1 patient showed increased sensitivity to DNA-damaging drugs. The findings suggested a link between this form of spastic paraplegia, which could be considered a neurodegenerative disease, and defects in DNA repair.</Attribute>\n",
      "              <Citation>\n",
      "                <ID Source=\"PubMed\">20613862</ID>\n",
      "              </Citation>\n",
      "              <XRef DB=\"OMIM\" ID=\"613647\" Type=\"MIM\" />\n",
      "            </ObservedData>\n",
      "          </ObservedIn>\n",
      "        </ObservedInList>\n",
      "        <SimpleAllele>\n",
      "          <GeneList>\n",
      "            <Gene Symbol=\"AP5Z1\" />\n",
      "          </GeneList>\n",
      "          <Name>AP5Z1, 4-BP DEL/22-BP INS, NT80</Name>\n",
      "          <VariantType>Variation</VariantType>\n",
      "          <OtherNameList>\n",
      "            <Name Type=\"NonHGVS\">4-BP DEL/22-BP INS, NT80</Name>\n",
      "          </OtherNameList>\n",
      "          <XRefList>\n",
      "            <XRef DB=\"OMIM\" ID=\"613653.0001\" Type=\"Allelic variant\" />\n",
      "          </XRefList>\n",
      "        </SimpleAllele>\n",
      "        <TraitSet Type=\"Disease\">\n",
      "          <Trait Type=\"Disease\">\n",
      "            <Name>\n",
      "              <ElementValue Type=\"Preferred\">SPASTIC PARAPLEGIA 48, AUTOSOMAL RECESSIVE</ElementValue>\n",
      "            </Name>\n",
      "          </Trait>\n",
      "        </TraitSet>\n",
      "      </ClinicalAssertion>\n",
      "      <ClinicalAssertion ID=\"2865972\" SubmissionDate=\"2020-11-14\" DateLastUpdated=\"2021-05-16\" DateCreated=\"2021-05-16\">\n",
      "        <ClinVarSubmissionID localKey=\"NM_014855.3:c.80_83delinsTGCTGTAAACTGTAACTGTAAA|OMIM:613647\" submittedAssembly=\"GRCh37\" />\n",
      "        <ClinVarAccession Accession=\"SCV001451119\" DateUpdated=\"2021-05-16\" DateCreated=\"2021-05-16\" Type=\"SCV\" Version=\"1\" SubmitterName=\"Paris Brain Institute, Inserm - ICM\" OrgID=\"507826\" OrganizationCategory=\"laboratory\" />\n",
      "        <RecordStatus>current</RecordStatus>\n",
      "        <Classification>\n",
      "          <ReviewStatus>criteria provided, single submitter</ReviewStatus>\n",
      "          <GermlineClassification>Pathogenic</GermlineClassification>\n",
      "        </Classification>\n",
      "        <Assertion>variation to disease</Assertion>\n",
      "        <AttributeSet>\n",
      "          <Attribute Type=\"AssertionMethod\">ACMG Guidelines, 2015</Attribute>\n",
      "          <Citation>\n",
      "            <ID Source=\"PubMed\">25741868</ID>\n",
      "          </Citation>\n",
      "        </AttributeSet>\n",
      "        <ObservedInList>\n",
      "          <ObservedIn>\n",
      "            <Sample>\n",
      "              <Origin>unknown</Origin>\n",
      "              <Species TaxonomyId=\"9606\">human</Species>\n",
      "              <AffectedStatus>yes</AffectedStatus>\n",
      "            </Sample>\n",
      "            <Method>\n",
      "              <MethodType>clinical testing</MethodType>\n",
      "            </Method>\n",
      "            <ObservedData>\n",
      "              <Attribute Type=\"VariantAlleles\" integerValue=\"2\" />\n",
      "            </ObservedData>\n",
      "          </ObservedIn>\n",
      "        </ObservedInList>\n",
      "        <SimpleAllele>\n",
      "          <VariantType>Variation</VariantType>\n",
      "          <AttributeSet>\n",
      "            <Attribute Type=\"HGVS\">NM_014855.3:c.80_83delinsTGCTGTAAACTGTAACTGTAAA</Attribute>\n",
      "          </AttributeSet>\n",
      "        </SimpleAllele>\n",
      "        <TraitSet Type=\"Disease\">\n",
      "          <Trait Type=\"Disease\">\n",
      "            <XRef DB=\"OMIM\" ID=\"613647\" Type=\"MIM\" />\n",
      "          </Trait>\n",
      "        </TraitSet>\n",
      "        <SubmissionNameList>\n",
      "          <SubmissionName>SUB8526155</SubmissionName>\n",
      "        </SubmissionNameList>\n",
      "      </ClinicalAssertion>\n",
      "    </ClinicalAssertionList>\n",
      "    <TraitMappingList>\n",
      "      <TraitMapping ClinicalAssertionID=\"20155\" TraitType=\"Disease\" MappingType=\"Name\" MappingValue=\"SPASTIC PARAPLEGIA 48, AUTOSOMAL RECESSIVE\" MappingRef=\"Preferred\">\n",
      "        <MedGen CUI=\"C3150901\" Name=\"Hereditary spastic paraplegia 48\" />\n",
      "      </TraitMapping>\n",
      "      <TraitMapping ClinicalAssertionID=\"2865972\" TraitType=\"Disease\" MappingType=\"XRef\" MappingValue=\"613647\" MappingRef=\"OMIM\">\n",
      "        <MedGen CUI=\"C3150901\" Name=\"Hereditary spastic paraplegia 48\" />\n",
      "      </TraitMapping>\n",
      "    </TraitMappingList>\n",
      "  </ClassifiedRecord>\n",
      "</VariationArchive>\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import xml.etree.ElementTree as ET\n",
    "import gzip\n",
    "\n",
    "# Define XML file\n",
    "xml_file = \"ClinVarFullRelease_2025-01.xml.gz\"\n",
    "\n",
    "# Define number of records to print\n",
    "N = 1\n",
    "\n",
    "# Stream process compressed XML file\n",
    "def print_first_n_records(xml_file, N):\n",
    "    with gzip.open(xml_file, \"rt\", encoding=\"utf-8\") as f:\n",
    "        context = ET.iterparse(f, events=(\"start\", \"end\"))\n",
    "        _, root = next(context)  # Get root element\n",
    "\n",
    "        record_count = 0\n",
    "        for event, elem in context:\n",
    "            if event == \"end\" and elem.tag in [\"ClinVarAssertion\", \"VariationArchive\"]:  \n",
    "                # Adjust tag name as needed based on your XML structure\n",
    "                print(ET.tostring(elem, encoding=\"unicode\"))\n",
    "                record_count += 1\n",
    "                elem.clear()  # Free memory\n",
    "                if record_count >= N:\n",
    "                    break\n",
    "\n",
    "# Call function to print first N records\n",
    "print_first_n_records(xml_file, N)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bioinfo",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
