{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from clinvar_functions import *\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import pysam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Path to ClinVar VCF file\n",
    "vcf_file = \"./data/clinvar.vcf\"\n",
    "\n",
    "# Open the VCF file\n",
    "vcf_reader = pysam.VariantFile(vcf_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Fields: ['ALLELEID',\n",
    " 'CLNDISDB',\n",
    " 'CLNDN',\n",
    " 'CLNHGVS',\n",
    " 'CLNREVSTAT',\n",
    " 'CLNSIG',\n",
    " 'CLNSIGSCV',\n",
    " 'CLNVC',\n",
    " 'CLNVCSO',\n",
    " 'CLNVI',\n",
    " 'GENEINFO',\n",
    " 'ORIGIN']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error processing record at 16:31092475:VKORC1:79001 - 'NoneType' object is not iterable\n",
      "Error processing record at 16:31096368:VKORC1:79001 - 'NoneType' object is not iterable\n",
      "(445, 10)\n"
     ]
    }
   ],
   "source": [
    "# Extract information from first 10 variants\n",
    "variants = []\n",
    "for record in vcf_reader.fetch():\n",
    "    gene_info = record.info.get(\"GENEINFO\", [\"\"])\n",
    "    if \"ORC1\" in gene_info:\n",
    "        try:\n",
    "            variant_info = {\n",
    "                \"CHROM\": record.chrom,  # Chromosome\n",
    "                \"POS\": record.pos,      # Position\n",
    "                \"ID\": record.id,        # Variant ID (e.g., rs number)\n",
    "                \"GENEINFO\":gene_info,   # Gene information\n",
    "                \"REF\": record.ref,      # Reference allele\n",
    "                \"ALT\": \",\".join(str(alt) for alt in record.alts),  # Alternate allele(s)\n",
    "                \"CLNREVSTAT\": record.info.get(\"CLNREVSTAT\"),  # Review status\n",
    "                \"CLNSIG\": record.info.get(\"CLNSIG\"),  # Clinical significance\n",
    "                \"RESULT\": record.info.get(\"MC\"),  # Molecular consequence\n",
    "                \"CLNDN\": record.info.get(\"CLNDN\")  # Disease name\n",
    "            }\n",
    "            variants.append(variant_info)\n",
    "        except Exception as e:\n",
    "            print(f\"Error processing record at {record.chrom}:{record.pos}:{gene_info} - {e}\")\n",
    "    \n",
    "\n",
    "# Convert to DataFrame\n",
    "df = pd.DataFrame(variants)\n",
    "print(df.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'MORC1:27136',\n",
       " 'ORC1:4998',\n",
       " 'ORC1:4998|LOC126805733:126805733',\n",
       " 'ORC1:4998|PRPF38A:84950',\n",
       " 'VKORC1:79001',\n",
       " 'VKORC1:79001|PRSS53:339105',\n",
       " 'VKORC1L1:154807'}"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# List of genes after filtration\n",
    "set(df['GENEINFO'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(339, 10)\n"
     ]
    }
   ],
   "source": [
    "# Filterded by gene name\n",
    "df = df[(df['GENEINFO']=='ORC1:4998|PRPF38A:84950') | (df['GENEINFO']=='ORC1:4998') | (df['GENEINFO']=='ORC1:4998|LOC126805733:126805733')]\n",
    "print(df.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Functions for data cleaning\n",
    "def convert_to_str(x):\n",
    "    if x is not None:\n",
    "        return x[0]\n",
    "    return 'not_specified'\n",
    "\n",
    "def second_element(x):\n",
    "    li = list(x)\n",
    "    if len(li) == 2:\n",
    "        return li[1][1:]\n",
    "    return 'not_specified'\n",
    "\n",
    "def split_row(row):\n",
    "    try:\n",
    "        return row.split(\"|\")[1]\n",
    "    except:\n",
    "        return row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"CLNSIG\"] = df[\"CLNSIG\"].apply(lambda x: convert_to_str(x))\n",
    "df[\"RESULT\"] = df[\"RESULT\"].apply(lambda x: convert_to_str(x))\n",
    "df[\"CLNDN\"] = df[\"CLNDN\"].apply(lambda x: convert_to_str(x))\n",
    "\n",
    "df[\"CRITERIA\"] = df[\"CLNREVSTAT\"].apply(lambda x: list(x)[0])\n",
    "df[\"SUBMITTION\"] = df[\"CLNREVSTAT\"].apply(lambda x: second_element(x))\n",
    "df[\"RESULT\"] = df[\"RESULT\"].apply(lambda x: split_row(x))\n",
    "\n",
    "#df = df.drop(columns=['CLNREVSTAT'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>RESULT</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CRITERIA</th>\n",
       "      <th>SUBMITTION</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>52369375</td>\n",
       "      <td>1173067</td>\n",
       "      <td>ORC1:4998</td>\n",
       "      <td>TGTACATCTCCCATATTGCAACACCTCCCTTTCCCAAAGATGTACC...</td>\n",
       "      <td>T</td>\n",
       "      <td>P</td>\n",
       "      <td>n</td>\n",
       "      <td>M</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>single_submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>52372950</td>\n",
       "      <td>297574</td>\n",
       "      <td>ORC1:4998</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>B</td>\n",
       "      <td>S</td>\n",
       "      <td>M</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>not_specified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>52372955</td>\n",
       "      <td>297575</td>\n",
       "      <td>ORC1:4998</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>U</td>\n",
       "      <td>S</td>\n",
       "      <td>M</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>single_submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>52373087</td>\n",
       "      <td>297576</td>\n",
       "      <td>ORC1:4998</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>U</td>\n",
       "      <td>S</td>\n",
       "      <td>M</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>single_submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>52373187</td>\n",
       "      <td>703217</td>\n",
       "      <td>ORC1:4998</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>S</td>\n",
       "      <td>M</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>conflicting_classifications</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334</th>\n",
       "      <td>1</td>\n",
       "      <td>52402240</td>\n",
       "      <td>875878</td>\n",
       "      <td>ORC1:4998</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>U</td>\n",
       "      <td>S</td>\n",
       "      <td>M</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>single_submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335</th>\n",
       "      <td>1</td>\n",
       "      <td>52404418</td>\n",
       "      <td>875879</td>\n",
       "      <td>ORC1:4998|LOC126805733:126805733</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>U</td>\n",
       "      <td>S</td>\n",
       "      <td>M</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>single_submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336</th>\n",
       "      <td>1</td>\n",
       "      <td>52404439</td>\n",
       "      <td>297590</td>\n",
       "      <td>ORC1:4998|LOC126805733:126805733</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>U</td>\n",
       "      <td>S</td>\n",
       "      <td>M</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>single_submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>337</th>\n",
       "      <td>1</td>\n",
       "      <td>52408585</td>\n",
       "      <td>3425738</td>\n",
       "      <td>ORC1:4998|PRPF38A:84950</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>U</td>\n",
       "      <td>S</td>\n",
       "      <td>n</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>single_submitter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338</th>\n",
       "      <td>1</td>\n",
       "      <td>52408613</td>\n",
       "      <td>3425737</td>\n",
       "      <td>ORC1:4998|PRPF38A:84950</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>U</td>\n",
       "      <td>S</td>\n",
       "      <td>n</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>single_submitter</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>339 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS       ID                          GENEINFO  \\\n",
       "0       1  52369375  1173067                         ORC1:4998   \n",
       "1       1  52372950   297574                         ORC1:4998   \n",
       "2       1  52372955   297575                         ORC1:4998   \n",
       "3       1  52373087   297576                         ORC1:4998   \n",
       "4       1  52373187   703217                         ORC1:4998   \n",
       "..    ...       ...      ...                               ...   \n",
       "334     1  52402240   875878                         ORC1:4998   \n",
       "335     1  52404418   875879  ORC1:4998|LOC126805733:126805733   \n",
       "336     1  52404439   297590  ORC1:4998|LOC126805733:126805733   \n",
       "337     1  52408585  3425738           ORC1:4998|PRPF38A:84950   \n",
       "338     1  52408613  3425737           ORC1:4998|PRPF38A:84950   \n",
       "\n",
       "                                                   REF ALT CLNSIG RESULT  \\\n",
       "0    TGTACATCTCCCATATTGCAACACCTCCCTTTCCCAAAGATGTACC...   T      P      n   \n",
       "1                                                    G   A      B      S   \n",
       "2                                                    C   G      U      S   \n",
       "3                                                    G   A      U      S   \n",
       "4                                                    G   A      C      S   \n",
       "..                                                 ...  ..    ...    ...   \n",
       "334                                                  T   A      U      S   \n",
       "335                                                  G   A      U      S   \n",
       "336                                                  G   C      U      S   \n",
       "337                                                  G   C      U      S   \n",
       "338                                                  C   T      U      S   \n",
       "\n",
       "    CLNDN           CRITERIA                   SUBMITTION  \n",
       "0       M  criteria_provided             single_submitter  \n",
       "1       M  criteria_provided                not_specified  \n",
       "2       M  criteria_provided             single_submitter  \n",
       "3       M  criteria_provided             single_submitter  \n",
       "4       M  criteria_provided  conflicting_classifications  \n",
       "..    ...                ...                          ...  \n",
       "334     M  criteria_provided             single_submitter  \n",
       "335     M  criteria_provided             single_submitter  \n",
       "336     M  criteria_provided             single_submitter  \n",
       "337     n  criteria_provided             single_submitter  \n",
       "338     n  criteria_provided             single_submitter  \n",
       "\n",
       "[339 rows x 11 columns]"
      ]
     },
     "execution_count": 183,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>RESULT</th>\n",
       "      <th>CLNDN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>52369375</td>\n",
       "      <td>1173067</td>\n",
       "      <td>TGTACATCTCCCATATTGCAACACCTCCCTTTCCCAAAGATGTACC...</td>\n",
       "      <td>T</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td></td>\n",
       "      <td>Meier-Gorlin_syndrome_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>52372950</td>\n",
       "      <td>297574</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Meier-Gorlin_syndrome_1|not_provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>52372955</td>\n",
       "      <td>297575</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Meier-Gorlin_syndrome_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>52373087</td>\n",
       "      <td>297576</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Meier-Gorlin_syndrome_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>52373187</td>\n",
       "      <td>703217</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>Meier-Gorlin_syndrome_1|not_provided|ORC1-rela...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>16</td>\n",
       "      <td>31094731</td>\n",
       "      <td>884454</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Vitamin_K-dependent_clotting_factors</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>16</td>\n",
       "      <td>31094774</td>\n",
       "      <td>884455</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Vitamin_K-dependent_clotting_factors</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>442</th>\n",
       "      <td>16</td>\n",
       "      <td>31094889</td>\n",
       "      <td>318978</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td></td>\n",
       "      <td>Vitamin_K-Dependent_Clotting_Factors</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>443</th>\n",
       "      <td>16</td>\n",
       "      <td>31096316</td>\n",
       "      <td>60671</td>\n",
       "      <td>A</td>\n",
       "      <td>AC</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>drug_response</td>\n",
       "      <td></td>\n",
       "      <td>Warfarin_response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>444</th>\n",
       "      <td>16</td>\n",
       "      <td>31096368</td>\n",
       "      <td>2211</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>reviewed_by_expert_panel</td>\n",
       "      <td>drug_response</td>\n",
       "      <td></td>\n",
       "      <td>Venous_thromboembolism|not_provided|See_cases|...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>445 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS       ID  \\\n",
       "0       1  52369375  1173067   \n",
       "1       1  52372950   297574   \n",
       "2       1  52372955   297575   \n",
       "3       1  52373087   297576   \n",
       "4       1  52373187   703217   \n",
       "..    ...       ...      ...   \n",
       "440    16  31094731   884454   \n",
       "441    16  31094774   884455   \n",
       "442    16  31094889   318978   \n",
       "443    16  31096316    60671   \n",
       "444    16  31096368     2211   \n",
       "\n",
       "                                                   REF ALT  \\\n",
       "0    TGTACATCTCCCATATTGCAACACCTCCCTTTCCCAAAGATGTACC...   T   \n",
       "1                                                    G   A   \n",
       "2                                                    C   G   \n",
       "3                                                    G   A   \n",
       "4                                                    G   A   \n",
       "..                                                 ...  ..   \n",
       "440                                                  A   G   \n",
       "441                                                  C   T   \n",
       "442                                                  C   G   \n",
       "443                                                  A  AC   \n",
       "444                                                  C   T   \n",
       "\n",
       "                         CLNREVSTAT  \\\n",
       "0                 criteria_provided   \n",
       "1                 criteria_provided   \n",
       "2                 criteria_provided   \n",
       "3                 criteria_provided   \n",
       "4                 criteria_provided   \n",
       "..                              ...   \n",
       "440               criteria_provided   \n",
       "441               criteria_provided   \n",
       "442               criteria_provided   \n",
       "443  no_assertion_criteria_provided   \n",
       "444        reviewed_by_expert_panel   \n",
       "\n",
       "                                           CLNSIG               RESULT  \\\n",
       "0                                      Pathogenic                        \n",
       "1                                          Benign  3_prime_UTR_variant   \n",
       "2                          Uncertain_significance  3_prime_UTR_variant   \n",
       "3                          Uncertain_significance  3_prime_UTR_variant   \n",
       "4    Conflicting_classifications_of_pathogenicity   synonymous_variant   \n",
       "..                                            ...                  ...   \n",
       "440                        Uncertain_significance  5_prime_UTR_variant   \n",
       "441                        Uncertain_significance  5_prime_UTR_variant   \n",
       "442                        Uncertain_significance                        \n",
       "443                                 drug_response                        \n",
       "444                                 drug_response                        \n",
       "\n",
       "                                                 CLNDN  \n",
       "0                              Meier-Gorlin_syndrome_1  \n",
       "1                 Meier-Gorlin_syndrome_1|not_provided  \n",
       "2                              Meier-Gorlin_syndrome_1  \n",
       "3                              Meier-Gorlin_syndrome_1  \n",
       "4    Meier-Gorlin_syndrome_1|not_provided|ORC1-rela...  \n",
       "..                                                 ...  \n",
       "440               Vitamin_K-dependent_clotting_factors  \n",
       "441               Vitamin_K-dependent_clotting_factors  \n",
       "442               Vitamin_K-Dependent_Clotting_Factors  \n",
       "443                                  Warfarin_response  \n",
       "444  Venous_thromboembolism|not_provided|See_cases|...  \n",
       "\n",
       "[445 rows x 9 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bioinfo",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
